pkg/schedule/checker/rule_checker.go

```diff
@@ -459,6 +459,13 @@ loopFits:
 				hasUnhealthyFit = true
 				break loopFits
 			}
+			// avoid to meet down store when fix orpahn peers,
+			// Isdisconnected is more strictly than IsUnhealthy.
+			if c.cluster.GetStore(p.GetStoreId()).IsDisconnected() {
+				hasUnhealthyFit = true
+				pinDownPeer = p
```

Comment:
if it's disconnected, we don't need to hurry to remove it.

Reply:
how about skipping the disconnected store when fixing orphan peers, only removing them when the store is down or the store is normal?

Reply:
forget about it, replacing the disconnect store's peer with orphan peer is reasonable.

Reply:
may meet issues like https://github.com/tikv/pd/issues/4045. we should make sure the replacement peer is healthy (orphan peer).

Reply:
these peers are in disconnect or down store, which they will change to down peer.

Reply:
@lhy1024 I mean make sure the remaining peer is health.

Reply:
I added check about disconnect store